buzz_el/entity_matcher/entity_matcher.py

```diff
@@ -0,0 +1,158 @@
+
+    Parameters
+    ----------
+    graph_loader : GraphLoader
+        An instance of the GraphLoader class for loading knowledge graph data.
+    spacy_model : Language
+        The spaCy language to use when constructing the spaCy span ruler.
+    span_ruler_config : Optional[Dict], optional
+        Configuration for the spaCy span ruler, by default {"phrase_matcher_attr": "LOWER"}.
+        See <https://spacy.io/api/spanruler#config>
+
+    Attributes
+    ----------
+    spacy_model : Language
+        The spaCy language to use when constructing the spaCy span ruler.
+    graph_loader : GraphLoader
+        An instance of the GraphLoader class for loading knowledge graph data.
+    _span_ruler_config : Dict
+        Configuration for the spaCy span ruler, by default {"phrase_matcher_attr": "LOWER"}.
+        See: <https://spacy.io/api/spanruler#config>
+    spacy_component : SpanRuler
+        The spaCy span ruler component for the knowledge graph entity matching.
+    """
+
+    def __init__(
+        self,
+        graph_loader: GraphLoader,
+        spacy_model: Language,
+        span_ruler_config: Optional[Dict] = None,
+    ) -> None:
+        """
+        Initialiser for the entity matcher.
+
+        Parameters
+        ----------
+        graph_loader : GraphLoader
+            An instance of the GraphLoader class for loading knowledge graph data.
+        spacy_model : Language
+            The spaCy language to use when constructing the spaCy span ruler.
+        span_ruler_config : Optional[Dict], optional
+            Configuration for the spaCy span ruler, by default {"phrase_matcher_attr": "LOWER"}.
+            See: <https://spacy.io/api/spanruler#config>
+        """
+        self.spacy_model = spacy_model
+        self.graph_loader = graph_loader
+
+        if span_ruler_config is None:
+            self._span_ruler_config = {"phrase_matcher_attr": "LOWER"}
+        else:
+            self._span_ruler_config = span_ruler_config
+
+        self.build_entity_ruler()  # set self.spacy_component
+
+    def __call__(self, doc: Doc) -> None:
+        """
+        Apply the entity matching to a spaCy doc.
+
+        Parameters
+        ----------
+        doc : Doc
+            The spaCy doc to process.
+        """
+        if self.spacy_component is None:
+            self.build_entity_ruler()
+
+        self.spacy_component(doc)
+
+    def pipe(self, docs: Iterable[Doc]) -> Iterable:
+        """
+        Apply the entity matching component to an iterable of spaCy docs.
+
+        Parameters
+        ----------
+        docs : Iterable[Doc]
+            An iterable of spaCy docs to process.
+
+        Returns
+        -------
+        Iterable
+            An iterable of processed spaCy docs.
+        """
+        if self.spacy_component is None:
+            self.build_entity_ruler()
+
+        doc_process_pipe = self.spacy_component.pipe(docs)
+
+        return doc_process_pipe
```

Comment:
why the return of pipe function is different from the call function ?

Reply:
Same as above, that is how spaCy function.